lib/client.js

```diff
@@ -1580,10 +1586,26 @@ async function writeIterable ({ body, client, request, socket, contentLength, he
     .on('close', onDrain)
     .on('drain', onDrain)
 
+  let bodyAsyncIterable = body
+
+  if (util.isFormDataLike(body)) {
+    if (nodeMajor < 16 || (nodeMajor > 16 && nodeMinor < 5)) {
+      throw new InvalidArgumentError('Form-Data bodies are only supported in node v16.5 and newer.')
+    }
```

Comment:
This will throw on 17.5, 18.5, 19.5, etc.

Reply:
good catch, fixed in 5169b98022b20c8dfc16fafdcaa707636e2df051